near-sdk-core/src/code_generator/impl_item_method_info.rs

```diff
@@ -202,37 +201,52 @@ impl ImplItemMethodInfo {
             // result_serializer,
             // is_init,
             method_type,
-            original_sig,
-            ..
+            original_sig,            ..
         } = attr_signature_info;
         let return_ident = quote! { -> near_sdk::PendingContractTx };
         let params = quote! {
             &self, #pat_type_list
         };
         let ident_str = format!("{}", ident.to_string());
-        let body = if matches!(method_type, MethodType::View) {
-            quote! {
-                near_sdk::PendingContractTx::new(&self.account_id, #ident_str, args, true)
-            }
+        let is_view = if matches!(method_type, MethodType::View) {
+            quote! {true}
         } else {
-            quote! {
-                near_sdk::PendingContractTx::new(&self.account_id, #ident_str, args, false)
-            }
+            quote! {false}
         };
-        let non_bindgen_attrs = non_bindgen_attrs.iter().fold(TokenStream2::new(), |acc, value| {
-            quote! {
-                #acc
-                #value
-            }
-        });
+        
+        let non_bindgen_attrs = non_bindgen_attrs
+            .iter()
+            .fold(TokenStream2::new(), |acc, value| {
+                quote! {
+                    #acc
+                    #value
+                }
+            });
         let Signature { generics, .. } = original_sig;
         quote! {
             #[cfg(not(target_arch = "wasm32"))]
             #non_bindgen_attrs
             pub fn #ident#generics(#params) #return_ident {
-                #json_args
-                #body
+                #serialize_args
+                near_sdk::PendingContractTx::new_from_bytes(&self.account_id, #ident_str, args, #is_view)
             }
         }
     }
 }
+
+fn json_serialize(attr_signature_info: &AttrSigInfo) -> TokenStream2 {
+  let args: TokenStream2 = attr_signature_info
+  .input_args()
+  .fold(None, |acc: Option<TokenStream2>, value| {
+      let ident = &value.ident;
+      let ident_str = format!("{}", ident.to_string());
```

Comment:
Also why do we need format here?

Reply:
This was the way I got `"ident": ident` for creating the json macro.

Reply:
Doesn't `ident.to_string()` already provides a `String`?